src/frameworks/index.ts

```diff
@@ -523,10 +523,9 @@ ${firebaseDefaults ? `__FIREBASE_DEFAULTS__=${JSON.stringify(firebaseDefaults)}\
       // TODO move to templates
       await writeFile(
         join(functionsDist, "server.js"),
-        `const { onRequest } = require('firebase-functions/v2/https');
-const server = import('firebase-frameworks');
-exports.ssr = onRequest((req, res) => server.then(it => it.handle(req, res)));
-`
+        `import { onRequest } from 'firebase-functions/v2/https';
+         const server = import('firebase-frameworks');
+         export const ssr = onRequest((req, res) => server.then(it => it.handle(req, res)));`
       );
```

Comment:
Agree with your comment on the PR that we will need some logic to retain the old CJS behavior as well.

Should be straightforward with something like `if (packageJson.type === 'module') ...`.

We could extract the snippets to `const`s, keep them as inline literals, or even bring them out into the `templates` folder like the `TODO` comment suggests, but I'll defer to @jamesdaniels on that one.

Reply:
I could see

```ts
const importStatements = packageJson.type === 'module' ?
  "import { onRequest } from 'firebase-functions/v2/https';" : 
  "const { onRequest } = require('firebase-functions/v2/https');"`
```

if we wanted to DRY it up

```ts
const importsFrom = [["onRequest"], "firebase-functions/v2/https"];
const importStatements = importsFrom.map(([imports, from]) => packageJson.type === "module" ? ... : ...);
...
```

Reply:
We'll need the same for the export, right? Might be easier/clearer to just repeat the whole block.

Reply:
SG, let's just repeat the block